numpy/lib/arraysetops.py

```diff
@@ -551,6 +551,50 @@ def in1d(ar1, ar2, assume_unique=False, invert=False):
     ar1 = np.asarray(ar1).ravel()
     ar2 = np.asarray(ar2).ravel()
 
+    # Convert booleans to uint8 so we can use the fast integer algorithm
+    if ar1.dtype == np.bool_:
+        ar1 = ar1.view(np.uint8)
+    if ar2.dtype == np.bool_:
+        ar2 = ar2.view(np.uint8)
+
+    # Check if we can use a fast integer algorithm:
+    integer_arrays = (np.issubdtype(ar1.dtype, np.integer) and
+                      np.issubdtype(ar2.dtype, np.integer))
+
+    # Make sure we won't encounter memory errors
```

Comment:
I'm not sure I understand the intent behind this comment.

Reply:
Actually this comment should be removed as I discovered the new algorithm actually lowers memory use (contrary to my assumption).